src/dstore/dstore_internal.h

```diff
@@ -456,6 +465,18 @@ struct dstore_ops {
 
 	/* This function returns block size */
 	ssize_t (*obj_get_bsize) (dstore_oid_t *oid);
+
+	/* DSAL.OP_INIT Interface.
+	 * This function creates a new unmap operation using the
+	 * given inputs.
+	 */
+	int (*io_trunc_op_init)(struct dstore_obj *dobj,
```

Comment:
I wonder if we really need to add one more interface function here.

We could try to change `dstore_io_vec` to support extents-only (no IO buffers). If no-IO-buffers io_vec does not hold the invariant, then `dstore_io_vec.flags` might help us with that: we can introduce a flag "NO_IO_DATA" and when `dstore_io_vec.flags & NO_IO_DATA` is set then the invariant check will be ignoring data buffers and theirs sizes.

What do you think about that?

Reply:
Thanks for guiding me. I was feeling confused at this particular section. I have used the flags as suggested. Please do the review and let me know if I have implemented correctly.